src/anemoi/datasets/create/functions/filters/single_level_relative_humidity_to_specific_humidity.py

```diff
@@ -0,0 +1,110 @@
+# Written by Michiel Van Ginderachter (mpvginde)
+# Date: 2024-06-27
+
+
+import numpy as np
+from earthkit.data.indexing.fieldlist import FieldArray
+from earthkit.meteo import thermo
+
+from anemoi.datasets.create.functions.filters.single_level_specific_humidity_to_relative_humidity import AutoDict
+from anemoi.datasets.create.functions.filters.single_level_specific_humidity_to_relative_humidity import NewDataField
+from anemoi.datasets.create.functions.filters.single_level_specific_humidity_to_relative_humidity import \
+    pressure_at_height_level
+
+
+def execute(context, input, height, t, rh, sp, new_name="2q", **kwargs):
+    """Convert the single (height) level relative humidity to specific humidity"""
+    result = FieldArray()
+
+    MANDATORY_KEYS = ["A", "B"]
+    OPTIONAL_KEYS = ["t_ml", "q_ml"]
+    MISSING_KEYS = []
+    DEFAULTS = dict(t_ml="t", q_ml="q")
+
+    for key in OPTIONAL_KEYS:
+        if key not in kwargs:
+            print(f"key {key} not found in yaml-file, using default key: {DEFAULTS[key]}")
```

Comment:
This `print` should probably be replaced by a log-message

Reply:
I'll do it.